backend/src/test/java/com/woowacourse/kkogkkog/domain/CouponTest.java

```diff
@@ -7,85 +7,115 @@
-                CouponStatus.READY);
-
-        coupon.changeStatus(CouponEvent.REQUEST, receiver);
-
-        assertThat(coupon.getCouponStatus()).isEqualTo(CouponStatus.REQUESTED);
+    @Nested
+    @DisplayName("생성자는")
+    class constructor {
+
+        @Test
+        @DisplayName("쿠폰의 정보를 받으면, 쿠폰을 생성한다.")
+        void success() {
+            Member sender = MemberFixture.ROOKIE;
+            Member receiver = MemberFixture.ARTHUR;
+
+            Coupon actual = new Coupon(null, sender, receiver, "한턱쏘는", "추가 메세지", "#241223", CouponType.COFFEE,
+                    CouponStatus.READY);
+
+            assertThat(actual).isNotNull();
+        }
+
+        @Test
+        @DisplayName("보낸 사람과 받는 사람이 동일하면, 예외를 던진다.")
+        void fail_sameSenderAndReceiver() {
+            Member member = MemberFixture.ROOKIE;
+            assertThatThrownBy(
+                    () -> new Coupon(null, member, member, "한턱쏘는", "추가 메세지", "#241223", CouponType.COFFEE,
+                            CouponStatus.READY)
+            ).isInstanceOf(SameSenderReceiverException.class);
+        }
     }
 
-    @Test
-    void 보낸_사람은_쿠폰_사용_요청을_보낼_수_없다() {
-        Member sender = MemberFixture.ROOKIE;
-        Member receiver = MemberFixture.ARTHUR;
-        Coupon coupon = new Coupon(null, sender, receiver, "한턱쏘는", "추가 메세지", "#241223", CouponType.COFFEE,
-                CouponStatus.READY);
-
-        assertThatThrownBy(() -> coupon.changeStatus(CouponEvent.REQUEST, sender))
-                .isInstanceOf(ForbiddenException.class);
-    }
-
-    @Test
-    void 받은_사람은_REQUESTED_상태의_쿠폰에_대한_사용_요청_취소를_보낼_수_있다() {
-        Member sender = MemberFixture.ROOKIE;
-        Member receiver = MemberFixture.ARTHUR;
-        Coupon coupon = new Coupon(null, sender, receiver, "한턱쏘는", "추가 메세지", "#241223", CouponType.COFFEE,
-                CouponStatus.REQUESTED);
-
-        coupon.changeStatus(CouponEvent.CANCEL, receiver);
-
-        assertThat(coupon.getCouponStatus()).isEqualTo(CouponStatus.READY);
-    }
-
-    @Test
-    void 받은_사람은_READY_상태의_쿠폰에_대한_사용_요청_취소를_보낼_수_없다() {
-        Member sender = MemberFixture.ROOKIE;
-        Member receiver = MemberFixture.ARTHUR;
-        Coupon coupon = new Coupon(null, sender, receiver, "한턱쏘는", "추가 메세지", "#241223", CouponType.COFFEE,
-                CouponStatus.READY);
-
-        assertThatThrownBy(() -> coupon.changeStatus(CouponEvent.CANCEL, receiver))
-                .isInstanceOf(InvalidRequestException.class);
-    }
-
-    @Test
-    void 보낸_사람은_쿠폰_사용_요청_취소를_보낼_수_없다() {
-        Member sender = MemberFixture.ROOKIE;
-        Member receiver = MemberFixture.ARTHUR;
-        Coupon coupon = new Coupon(null, sender, receiver, "한턱쏘는", "추가 메세지", "#241223", CouponType.COFFEE,
-                CouponStatus.REQUESTED);
-
-        assertThatThrownBy(() -> coupon.changeStatus(CouponEvent.CANCEL, sender))
-                .isInstanceOf(ForbiddenException.class);
+    @Nested
+    @DisplayName("changeStatus 메서드는")
+    class ChangeStatus {
```

Comment:
테스트 클래스 명칭 컨벤션에 위배된다는 인텔리제이 경고가 전체적으로 여러곳에서 보입니다.

<img width="642" alt="image" src="https://user-images.githubusercontent.com/73531614/180632172-23466390-e6b0-43cd-b8a2-fb11fa6874d5.png">

```suggestion
    @Nested
    @DisplayName("changeStatus 메서드는")
    class ChangeStatusTest {
```

Reply:
https://youtrack.jetbrains.com/issue/IDEA-284162
https://junit.org/junit5/docs/current/user-guide/#writing-tests-nested

관련 이슈에 대한 링크입니다.
IntelliJ에서 경고를 띄우는게 불편하다면 Test 추가하는 것도 방법이긴한데, 개인적으론 없는게 나아보이긴합니다.

Reply:
방금 확인해봤는데 저는 경고가 안나오긴 합니다. 인텔리제이를 업그레이드 하면 안나오는 것 같습니다.